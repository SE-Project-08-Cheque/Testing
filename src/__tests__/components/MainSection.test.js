// import dependencies
import React from 'react';
// import react dom
import ReactDOM from 'react-dom';
// import react-testing methods
import {render, cleanup, fireEvent} from '@testing-library/react';
// add custom jest matchers from jest-dom
import "@testing-library/jest-dom/extend-expect";
// import renderer for take snapshots
import renderer from 'react-test-renderer';

// import the component for testing
import MainSection from '../../components/MainSection';
import { MemoryRouter } from 'react-router-dom';


afterEach(cleanup);

it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<MemoryRouter><MainSection /></MemoryRouter>, div)
})

it("renders main section correctly", () => {
    const {getByTestId} = render(<MemoryRouter><MainSection /></MemoryRouter>)
    expect(getByTestId('welcome')).toBeTruthy();
    expect(getByTestId('welcome')).toHaveTextContent("Welcome");
    expect(getByTestId('send_cheque')).toBeInTheDocument();
    expect(getByTestId('send_cheque')).toHaveTextContent("Send a cheque");
    expect(getByTestId('button')).toBeInTheDocument();
    expect(getByTestId('button')).toHaveTextContent("Click Here");

})
it("click button correctly", () => {
    const {getByTestId} = render(<MemoryRouter><MainSection /></MemoryRouter>)
    fireEvent.click(getByTestId('button'));
})

it("matches snapshot", () => {
    const tree = renderer.create(<MemoryRouter><MainSection /></MemoryRouter>).toJSON();
    expect(tree).toMatchSnapshot();
})