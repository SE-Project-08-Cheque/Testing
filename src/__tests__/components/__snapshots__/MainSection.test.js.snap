// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
Array [
  <div
    className="main-section"
  >
    <div
      className="css-2u4u1k"
    >
      <div
        className="css-gmuwbf"
      >
        <p
          className="chakra-text css-17oyikl"
          data-testid="welcome"
        >
          Welcome!
        </p>
      </div>
      <div
        className="css-gmuwbf"
      >
        <p
          className="chakra-text css-7fdwc5"
          data-testid="send_cheque"
        >
          Send a cheque?
        </p>
      </div>
      <div
        className="css-gmuwbf"
      >
        <button
          className="chakra-button css-1w56fgd"
          data-testid="button"
          type="button"
        >
          Click Here!
          <span
            className="chakra-button__icon css-1oeb2oe"
          >
            <svg
              aria-hidden={true}
              fill="currentColor"
              focusable={false}
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M2.01 21L23 12 2.01 3 2 10l15 2-15 2z"
              />
            </svg>
          </span>
        </button>
      </div>
    </div>
  </div>,
  <div
    className="buttons"
    data-testid="cards"
    p="10px"
  >
    <div
      className="css-1aoxqvw"
    >
      <div
        className="css-19fmqat"
      >
        <div
          className="css-1txosv4"
        >
          <div
            className="css-gmuwbf"
          >
            <img
              alt="User"
              className="chakra-image__placeholder css-eweyq5"
            />
          </div>
        </div>
        <div
          className="css-1y4dbot"
        >
          <div
            className="css-n16x53"
          >
            <div
              className="css-x1sij0"
            >
              <span
                className="chakra-badge css-637omx"
              >
                Secured
              </span>
            </div>
            <div
              className="css-1mwm0bw"
            >
              <h2
                className="chakra-heading css-0"
              >
                Received Cheques
              </h2>
            </div>
            <div
              className="css-15yujhg"
            >
              <span
                className="css-ekuaes"
              />
            </div>
          </div>
        </div>
        <div
          className="css-dfiqwb"
        >
          <a
            href=""
            onClick={[Function]}
          >
            <button
              className="chakra-button css-16sbcsi"
              data-testid="go"
              type="button"
            >
              Go
              <span
                className="chakra-button__icon css-1oeb2oe"
              >
                <svg
                  aria-hidden={true}
                  fill="currentColor"
                  focusable={false}
                  height="1em"
                  stroke="currentColor"
                  strokeWidth="0"
                  style={
                    Object {
                      "color": undefined,
                    }
                  }
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M2.01 21L23 12 2.01 3 2 10l15 2-15 2z"
                  />
                </svg>
              </span>
            </button>
          </a>
        </div>
      </div>
    </div>
    <div
      className="css-1aoxqvw"
    >
      <div
        className="css-19fmqat"
      >
        <div
          className="css-1txosv4"
        >
          <div
            className="css-gmuwbf"
          >
            <img
              alt="User"
              className="chakra-image__placeholder css-eweyq5"
            />
          </div>
        </div>
        <div
          className="css-1y4dbot"
        >
          <div
            className="css-n16x53"
          >
            <div
              className="css-x1sij0"
            >
              <span
                className="chakra-badge css-637omx"
              >
                Secured
              </span>
            </div>
            <div
              className="css-1mwm0bw"
            >
              <h2
                className="chakra-heading css-0"
              >
                Sent Cheques
              </h2>
            </div>
            <div
              className="css-15yujhg"
            >
              <span
                className="css-ekuaes"
              />
            </div>
          </div>
        </div>
        <div
          className="css-dfiqwb"
        >
          <a
            href=""
            onClick={[Function]}
          >
            <button
              className="chakra-button css-16sbcsi"
              data-testid="go"
              type="button"
            >
              Go
              <span
                className="chakra-button__icon css-1oeb2oe"
              >
                <svg
                  aria-hidden={true}
                  fill="currentColor"
                  focusable={false}
                  height="1em"
                  stroke="currentColor"
                  strokeWidth="0"
                  style={
                    Object {
                      "color": undefined,
                    }
                  }
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M2.01 21L23 12 2.01 3 2 10l15 2-15 2z"
                  />
                </svg>
              </span>
            </button>
          </a>
        </div>
      </div>
    </div>
    <div
      className="css-1aoxqvw"
    >
      <div
        className="css-19fmqat"
      >
        <div
          className="css-1txosv4"
        >
          <div
            className="css-gmuwbf"
          >
            <img
              alt="User"
              className="chakra-image__placeholder css-eweyq5"
            />
          </div>
        </div>
        <div
          className="css-1y4dbot"
        >
          <div
            className="css-n16x53"
          >
            <div
              className="css-x1sij0"
            >
              <span
                className="chakra-badge css-637omx"
              >
                Secured
              </span>
            </div>
            <div
              className="css-1mwm0bw"
            >
              <h2
                className="chakra-heading css-0"
              >
                Finished Cheques
              </h2>
            </div>
            <div
              className="css-15yujhg"
            >
              <span
                className="css-ekuaes"
              />
            </div>
          </div>
        </div>
        <div
          className="css-dfiqwb"
        >
          <a
            href=""
            onClick={[Function]}
          >
            <button
              className="chakra-button css-16sbcsi"
              data-testid="go"
              type="button"
            >
              Go
              <span
                className="chakra-button__icon css-1oeb2oe"
              >
                <svg
                  aria-hidden={true}
                  fill="currentColor"
                  focusable={false}
                  height="1em"
                  stroke="currentColor"
                  strokeWidth="0"
                  style={
                    Object {
                      "color": undefined,
                    }
                  }
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M2.01 21L23 12 2.01 3 2 10l15 2-15 2z"
                  />
                </svg>
              </span>
            </button>
          </a>
        </div>
      </div>
    </div>
  </div>,
]
`;
